src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -514,7 +514,8 @@ protected internal void OnAnalyzerConfigDocumentOpened(DocumentId documentId, So
 
                 var newSolution = this.SetCurrentSolution(currentSolution);
 
-                SignupForTextChanges(documentId, textContainer, isCurrentContext, onDocumentTextChanged);
+                // PERF: Use a 5-second delay to process text changes to additional files and analyzer config files.
```

Comment:
Would this cause the workspace to get out of sync with the current buffers then? I'd be pretty worried if that has downstream impacts.

Reply:
This is absolutely a question where the answer also has to come from you :-) I have no idea, but trust your call - if you feel this is risky then we can take a different approach.

Reply:
My first thought was that this was terrifyingly risky. But we already dealt with races like this in the regular source files with our GetDocumentInCurrentContextWithChanges...and we don't have an equivalent for anything else. So I suspect either:

1. Nothing was assuming this today, so this is fine.
2. Something is assuming this today but was possibly already busted.

And if it's 2, the fix would be to perhaps introduce a WithChanges to fix it either way, but that might undo this optimization then.